Update SDL2 port

Daft-Freak opened the pull request:
SDL2 changes:
 - Update to upstream 2.0.7
 - Support `SIZEALL` cursor

Also fixes a test breakage caused by an upstream fix (https://github.com/emscripten-ports/SDL2/commit/fbd4f55982d234c47ac538b9cf3f12b8604b8ab2)

kripken commented on tests/sdl2_mouse.c:
```diff
@@ -17,10 +17,10 @@ void one() {
-        assert(eq(m->x, 10) && eq(m->y, 20) && eq(m->xrel, 10) && eq(m->yrel, 20)
+        assert(eq(m->x, 10) && eq(m->y, 20) && eq(m->xrel, 0) && eq(m->yrel, 0)
           || eq(m->x, 30) && eq(m->y, 70) && eq(m->xrel, 20) && eq(m->yrel, 50));
 #endif
```
If I understand this correctly, the xrel/yrel are 0 because it's the first motion? Makes sense, but then I think it would be good to explicitly test multiple motions, that is, note how many motions we saw before, and see that we get relative values of 0 just for the first.

Daft-Freak commented:
I've changed the test to track the number of motion events and added a comment about the first relative motion being zero. Also noticed a new file that needed to be built.

kripken commented:
Perfect, thanks @Daft-Freak!
